cloud/endagaweb/views/dashboard.py

```diff
@@ -688,6 +689,12 @@ def _handle_request(self, request):
             request.session['end_date'] = request.POST.get('end_date', None)
             request.session['services'] = request.POST.getlist('services[]',
                                                                None)
+            # Added to check password to download the csv
```

Comment:
Maybe we want to put this in the `ActivityView` instead? Otherwise all `ProtectedView` instances like `SubscriberInfo`, `SubscriberActivity`, etc all require this password in POST